repos/system_upgrade/cloudlinux/actors/clmysqlrepositorysetup/libraries/clmysqlrepositorysetup.py

```
@@ -181,18 +186,23 @@ def mariadb_process(self, repofile_name, repofile_data):
 
         Versions of MariaDB installed from https://mariadb.org/.
         """
-        cl8_repofile_list = []
+        cl_target_repofile_list = []
+        target_major = get_target_major_version()
+        source_major = get_source_major_version()
 
         for source_repo in repofile_data.data:
             # Maria URLs look like this:
             # baseurl = https://archive.mariadb.org/mariadb-10.3/yum/centos/7/x86_64
             # baseurl = https://archive.mariadb.org/mariadb-10.7/yum/centos7-ppc64/
-            # We want to replace the 7 in OS name after /yum/
+            # We want to replace the source_major in OS name after /yum/ with target_major
             target_repo = copy.deepcopy(source_repo)
-            target_repo.repoid = target_repo.repoid + "-8"
-            url_parts = target_repo.baseurl.split("yum")
-            url_parts[1] = "yum" + url_parts[1].replace("7", "8")
-            target_repo.baseurl = "".join(url_parts)
+            target_repo.repoid = "{}-{}".format(target_repo.repoid, target_major)
+            # Replace the first occurrence of source_major with target_major after 'yum'
+            url_parts = target_repo.baseurl.split("yum", 1)
+            if len(url_parts) == 2:
```

Should we report an error or a warning message here if split didn't go as expected? For example, in case of URL change.